src/workload.py

```diff
@@ -51,8 +51,9 @@ def alive(self) -> bool:
             return False
 
     @override
-    def write(self, content: str, path: str) -> None:
-        os.makedirs(os.path.dirname(path), exist_ok=True)
+    def write_file(self, content: str, file: str) -> None:
+        path = "/".join(file.split("/")[:-1])
```

Comment:
It would be better if you create the Path object first thing then manage it though the pathlib interface.
See [Path.parents](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents)

Reply:
Thanks for the hint! I like that, it's even cleaner.